### PR TITLE
feat: added json_array

### PR DESCRIPTION
**Pull request by akoshchiy**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Related #11270

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12665)

### CODE REVIEW

File: src/query/functions/src/scalars/variant.rs [hunk 1]
```diff
@@ -1057,3 +1081,25 @@ fn json_object_fn(
...
+    let len = args.iter().find_map(|arg| match arg {
+        ValueRef::Column(col) => Some(col.len()),
+        _ => None,
+    });
+    let mut columns = Vec::with_capacity(args.len());
+    for (i, arg) in args.iter().enumerate() {
+        let column = match arg {
+            ValueRef::Column(column) => column.clone(),
+            ValueRef::Scalar(s) => {
+                let column_builder = ColumnBuilder::repeat(s, 1, &ctx.generics[i]);
```
Comment (b41sh) [inline src/query/functions/src/scalars/variant.rs:1098]: The `num_rows` should be the length of the column. Sorry, this is an existing bug.

Comment (akoshchiy) [reply src/query/functions/src/scalars/variant.rs:1098]: Did you mean that it should be? 
`
ColumnBuilder::repeat(s, len.unwrap_or(1), &ctx.generics[i])
`

Comment (b41sh) [reply src/query/functions/src/scalars/variant.rs:1098]: yes

**Review by b41sh — APPROVED** [review]

**Comment by BohuTANG** [comment]
Thank you for the contribution!

**Comment by BohuTANG** [comment]
Looking forward to the documentation of this new function of JSON cc @soyeric128.

**Pull request merged**